### PR TITLE
Refactor python-hy into hylang-sh

### PR DESCRIPTION
**Pull request by tianon**
This allows `python-hy` to reference a separate `hylang-sh` test which is then applied to `hylang`.  Two birds, one stone.

@paultag you might like this one :+1:

### CODE REVIEW

File: test/tests/hylang-sh/container.hy [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  (print))
+
+(import [pip])
+(pip.main ["install" "-q" "sh"])
+
+(import [sh [echo]])
+(->
+  (+
+    (echo "-n" 21)
+    (echo "-n" 21))
```
Comment (tianon) [inline test/tests/hylang-sh/container.hy]: Whoops, this breaks.  Trying to make a slightly more interesting test.

Comment (tianon) [inline test/tests/hylang-sh/container.hy]: Fixed! :+1:

**Comment by tianon** [comment]
Currently running `bashbrew list python pypy hylang | xargs ./test/run.sh -t python-hy -t hylang-sh` BTW.

**Comment by tianon** [comment]
Interesting :smile:
```console
$ bashbrew list python pypy hylang | xargs ./test/run.sh -t python-hy -t hylang-sh
testing python:2.7.9
	'python-hy' [1/1]...passed
testing python:2.7
	'python-hy' [1/1]...passed
testing python:2
	'python-hy' [1/1]...passed
testing python:2.7.9-onbuild
	'python-hy' [1/1]...passed
testing python:2.7-onbuild
	'python-hy' [1/1]...passed
testing python:2-onbuild
	'python-hy' [1/1]...passed
testing python:2.7.9-slim
	'python-hy' [1/1]...passed
testing python:2.7-slim
	'python-hy' [1/1]...passed
testing python:2-slim
	'python-hy' [1/1]...passed
testing python:2.7.9-wheezy
	'python-hy' [1/1]...passed
testing python:2.7-wheezy
	'python-hy' [1/1]...passed
testing python:2-wheezy
	'python-hy' [1/1]...passed
testing python:3.3.6
	'python-hy' [1/1]...passed
testing python:3.3
	'python-hy' [1/1]...passed
testing python:3.3.6-onbuild
	'python-hy' [1/1]...passed
testing python:3.3-onbuild
	'python-hy' [1/1]...passed
testing python:3.3.6-slim
	'python-hy' [1/1]...passed
testing python:3.3-slim
	'python-hy' [1/1]...passed
testing python:3.3.6-wheezy
	'python-hy' [1/1]...passed
testing python:3.3-wheezy
	'python-hy' [1/1]...passed
testing python:3.4.3
	'python-hy' [1/1]...passed
testing python:3.4
	'python-hy' [1/1]...passed
testing python:3
	'python-hy' [1/1]...passed
testing python:latest
	'python-hy' [1/1]...passed
testing python:3.4.3-onbuild
	'python-hy' [1/1]...passed
testing python:3.4-onbuild
	'python-hy' [1/1]...passed
testing python:3-onbuild
	'python-hy' [1/1]...passed
testing python:onbuild
	'python-hy' [1/1]...passed
testing python:3.4.3-slim
	'python-hy' [1/1]...passed
testing python:3.4-slim
	'python-hy' [1/1]...passed
testing python:3-slim
	'python-hy' [1/1]...passed
testing python:slim
	'python-hy' [1/1]...passed
testing python:3.4.3-wheezy
	'python-hy' [1/1]...passed
testing python:3.4-wheezy
	'python-hy' [1/1]...passed
testing python:3-wheezy
	'python-hy' [1/1]...passed
testing python:wheezy
	'python-hy' [1/1]...passed
testing pypy:2-2.5.0
	'python-hy' [1/1]...failed; unexpected output:
--- /home/tianon/docker/stackbrew/test/tests/python-hy/expected-std-out.txt	2015-03-06 16:15:04.000000000 -0700
+++ -	2015-03-06 16:48:23.598609300 -0700
@@ -1,2 +1,6 @@
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 4
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 42
testing pypy:2-2.5
	'python-hy' [1/1]...failed; unexpected output:
--- /home/tianon/docker/stackbrew/test/tests/python-hy/expected-std-out.txt	2015-03-06 16:15:04.000000000 -0700
+++ -	2015-03-06 16:48:36.971036339 -0700
@@ -1,2 +1,6 @@
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 4
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 42
testing pypy:2-2
	'python-hy' [1/1]...failed; unexpected output:
--- /home/tianon/docker/stackbrew/test/tests/python-hy/expected-std-out.txt	2015-03-06 16:15:04.000000000 -0700
+++ -	2015-03-06 16:48:50.907354040 -0700
@@ -1,2 +1,6 @@
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 4
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 42
testing pypy:2
	'python-hy' [1/1]...failed; unexpected output:
--- /home/tianon/docker/stackbrew/test/tests/python-hy/expected-std-out.txt	2015-03-06 16:15:04.000000000 -0700
+++ -	2015-03-06 16:49:03.820897407 -0700
@@ -1,2 +1,6 @@
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 4
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 42
testing pypy:2-2.5.0-onbuild
	'python-hy' [1/1]...failed; unexpected output:
--- /home/tianon/docker/stackbrew/test/tests/python-hy/expected-std-out.txt	2015-03-06 16:15:04.000000000 -0700
+++ -	2015-03-06 16:49:17.704388611 -0700
@@ -1,2 +1,6 @@
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 4
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 42
testing pypy:2-2.5-onbuild
	'python-hy' [1/1]...failed; unexpected output:
--- /home/tianon/docker/stackbrew/test/tests/python-hy/expected-std-out.txt	2015-03-06 16:15:04.000000000 -0700
+++ -	2015-03-06 16:49:30.177650094 -0700
@@ -1,2 +1,6 @@
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 4
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 42
testing pypy:2-2-onbuild
	'python-hy' [1/1]...failed; unexpected output:
--- /home/tianon/docker/stackbrew/test/tests/python-hy/expected-std-out.txt	2015-03-06 16:15:04.000000000 -0700
+++ -	2015-03-06 16:49:42.916702730 -0700
@@ -1,2 +1,6 @@
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 4
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 42
testing pypy:2-onbuild
	'python-hy' [1/1]...failed; unexpected output:
--- /home/tianon/docker/stackbrew/test/tests/python-hy/expected-std-out.txt	2015-03-06 16:15:04.000000000 -0700
+++ -	2015-03-06 16:49:55.797980881 -0700
@@ -1,2 +1,6 @@
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 4
+You are using pip version 6.0.7, however version 6.0.8 is available.
+You should consider upgrading via the 'pip install --upgrade pip' command.
 42
testing pypy:2-2.5.0-slim
	'python-hy' [1/1]...passed
testing pypy:2-2.5-slim
	'python-hy' [1/1]...passed
testing pypy:2-2-slim
	'python-hy' [1/1]...passed
testing pypy:2-slim
	'python-hy' [1/1]...passed
testing pypy:3-2.4.0
	'python-hy' [1/1]...passed
testing pypy:3-2.4
	'python-hy' [1/1]...passed
testing pypy:3-2
	'python-hy' [1/1]...passed
testing pypy:3
	'python-hy' [1/1]...passed
testing pypy:latest
	'python-hy' [1/1]...passed
testing pypy:3-2.4.0-onbuild
	'python-hy' [1/1]...passed
testing pypy:3-2.4-onbuild
	'python-hy' [1/1]...passed
testing pypy:3-2-onbuild
	'python-hy' [1/1]...passed
testing pypy:3-onbuild
	'python-hy' [1/1]...passed
testing pypy:onbuild
	'python-hy' [1/1]...passed
testing pypy:3-2.4.0-slim
	'python-hy' [1/1]...passed
testing pypy:3-2.4-slim
	'python-hy' [1/1]...passed
testing pypy:3-2-slim
	'python-hy' [1/1]...passed
testing pypy:3-slim
	'python-hy' [1/1]...passed
testing pypy:slim
	'python-hy' [1/1]...passed
testing hylang:latest
	'hylang-sh' [1/1]...passed
testing hylang:0
	'hylang-sh' [1/1]...passed
testing hylang:0.10
	'hylang-sh' [1/1]...passed
testing hylang:0.10.1
	'hylang-sh' [1/1]...passed
```

**Comment by yosifkit** [comment]
Other than pypy failing, LGTM

**Comment by tianon** [comment]
which is "fixed" by a rebuild...  tried to find a better solution for that but I'm empty-handed ATM :disappointed:

**Comment by yosifkit** [comment]
ok, sounds fine for now.

**Pull request merged**